dbms/src/Storages/DeltaMerge/DMVersionFilterBlockInputStream.cpp

```diff
@@ -319,7 +322,11 @@ Block DMVersionFilterBlockInputStream<MODE>::read(FilterPtr & res_filter, bool r
             // Now let's handle the last row of current block.
             auto cur_handle = rowkey_column->getRowKeyValue(rows - 1);
             auto cur_version = (*version_col_data)[rows - 1];
-            auto deleted = (*delete_col_data)[rows - 1];
+            UInt8 deleted = 0;
+            if (delete_col_data != nullptr)
+            {
+                deleted = (*delete_col_data)[rows - 1];
+            }
```

Comment:
Using some bit art to avoid branch switches can improve performance normally.

```suggestion
            // UInt8 deleted = delete_col_data == nullptr ? 0 : (*delete_col_data)[rows - 1];
            UInt8 deleted = (UInt8) (delete_col_data != nullptr) & (*delete_col_data)[rows - 1];
```